app/code/Magento/Tax/Block/Sales/Order/Tax.php

```diff
@@ -84,7 +86,11 @@ public function initTotals()
         $allowTax = $this->_source->getTaxAmount() > 0 || $this->_config->displaySalesZeroTax($store);
         $grandTotal = (double)$this->_source->getGrandTotal();
         if (!$grandTotal || $allowTax && !$this->_config->displaySalesTaxWithGrandTotal($store)) {
-            $this->_addTax();
+            $taxTotal = new \Magento\Framework\DataObject(['code' => 'tax', 'block_name' => $this->getNameInLayout()]);
```

Comment:
I think it's better to move this login to `_addTax` method. `_addTax` method should be called from initTotals as third party code can rely on this call